### PR TITLE
fix(logging): RHICOMPL-1485 sidekiq context evidence for ver <6

### PR DESCRIPTION
**Pull request by vkrizan**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [X] Input Validation
- [X] Output Encoding
- [X] Authentication and Password Management
- [X] Session Management
- [X] Access Control
- [X] Cryptographic Practices
- [X] Error Handling and Logging
- [X] Data Protection
- [X] Communication Security
- [X] System Configuration
- [X] Database Security
- [X] File Management
- [X] Memory Management
- [X] General Coding Practices

### CODE REVIEW

File: lib/audit_log/audit_log/formatter.rb [hunk 1]
```diff
@@ -56,12 +56,19 @@ def sidekiq_current_ctx
...
             end
-          rescue NoMethodError
+          rescue StandardError
             nil
           end
 
+          def parse_sidekiq_ctx(ctx)
+            return unless ctx
+
+            curr_ctx = ctx[-1]
```
Comment (skateman) [inline lib/audit_log/audit_log/formatter.rb:68]: Why `-1`?

Comment (vkrizan) [reply lib/audit_log/audit_log/formatter.rb:68]: It seems that sidekiq appends every context, and we want to use the latest/effective one. Check https://github.com/mperham/sidekiq/blob/v5.2.7/lib/sidekiq/logging.rb#L47

`.last` might be used instead...

Comment (skateman) [reply lib/audit_log/audit_log/formatter.rb:68]: Yeah, `.last` is more readable

Comment (vkrizan) [reply lib/audit_log/audit_log/formatter.rb]: Updated to `last`.

**Review by skateman — APPROVED** [review]
The Seal of Approval
Verified that it fixes the issue :+1:

**Pull request merged**